pkgs/by-name/po/poco/package.nix

```diff
@@ -54,25 +55,54 @@ stdenv.mkDerivation rec {
   MYSQL_DIR = libmysqlclient;
   MYSQL_INCLUDE_DIR = "${MYSQL_DIR}/include/mysql";
 
-  cmakeFlags = [
-    # use nix provided versions of sqlite, zlib, pcre, expat, ... instead of bundled versions
-    (lib.cmakeBool "POCO_UNBUNDLED" true)
-  ];
+  cmakeFlags =
+    let
+      excludeTestsRegex = lib.concatStringsSep "|" [
+        # These tests require running services, which the checkPhase is ill equipeed to provide
+        # TODO get them running in a nixosTest
+        "Redis"
+        "DataODBC"
+        "MongoDB"
+        "DataMySQL"
+        # network not accessible from nix sandbox
+        "NetSSL" # around 25 test failures
+        "Net" # could be made to work when public network access is patched out
+      ];
+    in
+    [
+      # use nix provided versions of sqlite, zlib, pcre, expat, ... instead of bundled versions
+      (lib.cmakeBool "POCO_UNBUNDLED" true)
+      (lib.cmakeBool "ENABLE_TESTS" true)
+      (lib.cmakeFeature "CMAKE_CTEST_ARGUMENTS" "--exclude-regex;'${excludeTestsRegex}'")
+    ];
+
+  patches =
+    [
+      # Remove on next release
+      (fetchpatch {
```

Comment:
optional: I have seen on some PR that `fetchpatch2` should now be used.

Reply:
[That's still so new, it's not yet documented](https://nixos.org/manual/nixpkgs/unstable/#fetchpatch). The only difference seems to be, that it uses a newer version of patchutils, to combat aha-breakage on GitHub. The Code does not make it clear when this might happen, but none the less that seems like a good idea.

Reply:
Yeah, also the fetchpatch was not actually added in this pull request, but merely a whitespace change. I kind of hope that once the automated change to fetchpatch2 is through, there will once again be only a fetchpatch and no two of them, as that is uneccessarily confusing.